File: answers/leetcode/Implement Stack using Queues/Implement Stack using Queues.py
class Stack(object):
    def __init__(self):
        """
        initialize your data structure here.
        """

        self.queue0 = []
        self.queue1 = []

    def push(self, x):
        """
        :type x: int
        :rtype: nothing
        """

        self.queue0.append(x)

    def pop(self):
        """
        :rtype: nothing
        """

        self.top()
        self.queue0.pop(0)

    def top(self):
        """
        :rtype: int
        """

        if 0 == len(self.queue0):
            while len(self.queue1):
                self.queue0.append(self.queue1.pop(0))
        while len(self.queue0) > 1:
            self.queue1.append(self.queue0.pop(0))
        return self.queue0[0]

    def empty(self):
        """
        :rtype: bool
        """

        return 0 == len(self.queue0) + len(self.queue1)

File: answers/leetcode/Implement Stack using Queues/Implement Stack using Queues.java
class MyStack {

    private Queue<Integer> inQueue;
    private Queue<Integer> outQueue;

    /** Initialize your data structure here. */
    public MyStack() {
        this.inQueue = new PriorityQueue<Integer>();
        this.outQueue = new PriorityQueue<Integer>();
    }

    /** Push element x onto stack. */
    public void push(int x) {
        this.inQueue.add(x);
    }

    /** Removes the element on top of the stack and returns that element. */
    public int pop() {
        this.moveToOutQueue();
        return this.inQueue.remove();
    }

    /** Get the top element. */
    public int top() {
        this.moveToOutQueue();
        return this.inQueue.peek();
    }

    /** Returns whether the stack is empty. */
    public boolean empty() {
        return this.inQueue.isEmpty() && this.outQueue.isEmpty();
    }

    private void moveToOutQueue() {
        if (this.inQueue.isEmpty()) {
            this.inQueue = new PriorityQueue<Integer>(this.outQueue);
            this.outQueue.clear();
        }
        while (this.inQueue.size() > 1) {
            this.outQueue.add(this.inQueue.remove());
        }
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */